### ISSUE TITLE
Localstack ignores index projection type

### ISSUE THREAD

**Issue by piotrkilczuk** [localstack/localstack#805]
If I create a Global Secondary Index on a DynamoDB table using the following options:

![screenshot from 2018-06-15 14-47-18](https://user-images.githubusercontent.com/365324/41471663-2f0e926a-70ac-11e8-8920-e5b0f08efe27.png)

and query using `Select='ALL_ATTRIBUTES'`:

```
dynamodb = boto3.resource('dynamodb')
table = dynamodb.Table('name')
table.query(
    TableName='name',
    IndexName='id-index',
    KeyConditionExpression=Key('id').eq(1234567890),
    Select='ALL_ATTRIBUTES'
)
```

localstack will return all the fields from the table.

However, using `boto3` against a live DynamoDB instance I would get the following error:

```
One or more parameter values were invalid: Select type ALL_ATTRIBUTES is not supported for global secondary index id-index because its projection type is not ALL
```

The behaviour should be consistent with AWS and localstack should error when trying to ` Select='ALL_ATTRIBUTES'` on an index with 'Include' projection.

**Comment by usmangani1** [comment]
@piotrkilczuk Thanks for reporting the issue.

Can you please help us with the ProjectionType which is actually being used for index creation ?

A basic aws cli command to or boto script which has been used for creating index will be helpful.Thanks

**Comment by piotrkilczuk** [comment]
It's been a while, but we're talking about `ProjectionType == INCLUDE`.

**Comment by whummer** [comment]
Thanks for the update @piotrkilczuk @usmangani1 . This should be fixed in #2840 - this change should be available in the `latest` version of the Docker image. Please report here if the problem persists. Thanks

**Comment by therightstuff** [comment]
Hi - I'm using the latest version of localstack and I've run into what feels (at the very least) related: if I create an global secondary index with a projection type 'ALL' then a query against the index returns all data as expected, but if I do so with projection type 'INCLUDE' and a list of non-key attributes those specified attributes are not returned in my queries.
